### PR TITLE
[rocjitsu] Apply async IOFFSET to LDS addresses

### LINKED ISSUE #8829: [rocJITsu][gfx1250] cluster async-to-LDS INST_OFFSET does not adjust LDS destination
`cluster_load_async_to_lds_*` applies its signed `INST_OFFSET` byte offset to
both the global source and LDS destination. rocJITsu currently applies it only
to the global side, so a valid nonzero-offset gfx1250 cluster load places the
right payload in the wrong LDS locations.

The attached standalone probe uses only a matching ROCm nightly SDK. It
compiles a `(2,1,1)` HIP cluster, verifies that LLVM emitted
`cluster_load_async_to_lds_b32 ... offset:4 scale_offset`, submits the public
extended AQL packet through HSA, and checks both participant-owned LDS results.
The current nightly produces 1/32 correct lanes in each workgroup. The attached
generator-owned patch produces 32/32 in both.

The patch is intentionally cluster-only: ordinary
`GLOBAL_LOAD_ASYNC_TO_LDS` retains its current LDS addressing rule. Coverage
includes all four cluster widths in the generated bodies, positive and negative
signed offsets in decoded state, and an end-to-end two-workgroup simulation
that exercises decode, global memory, multicast remapping, ASYNCCNT completion,
and recipient LDS contents.

Reproducer and isolated patch:
https://gist.github.com/benvanik/c8c515b87391bfa34960ad3d81707fe2

### PR DESCRIPTION
**Pull request by kuhar**
Apply the signed 24-bit IOFFSET to both the global and LDS addresses for ordinary and cluster async loads and async stores, matching CDNA5 section 10.8.1 and expressions 95-102 and 106-109.

Reject LDS destinations outside the issuing workgroup allocation, keep cluster remapping widened through validation, and filter invalid lanes from writeback and race detection.

Add generator, execution, multicast, and plugin coverage for signed offsets, allocation bounds, and wraparound addresses.

Fixes #8829

### CODE REVIEW

File: emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/isa/arch/amdgpu/cdna5/addr_calc.cpp [hunk 1]
```diff
@@ -176,12 +177,24 @@ void flat_calculate_addresses(const VglobalMachineInst &inst, amdgpu::Wavefront
   flat_global_calculate_addresses(inst, wf, d, false);
 }
 
+uint32_t async_lds_lane_address(const VglobalMachineInst &inst, const amdgpu::Wavefront &wf,
+                                uint32_t lds_operand) {
+  int64_t relative_addr = static_cast<int64_t>(lds_operand) + signed_ioffset(inst.ioffset);
+  if (relative_addr < 0 || static_cast<uint64_t>(relative_addr) >= wf.lds_size())
```
Comment (atgutier) [inline emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/isa/arch/amdgpu/cdna5/addr_calc.cpp]: Since we're doing this runtime check we may also want to check the size of the request too to ensure it doesn't exceed for any byte address: `static_cast<uint64_t>(relative_addr) + request_size >= wf.lds_size().

**Review by atgutier — APPROVED** [review]
LGTM. Just one comment that could harden our bounds check.

File: emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/isa/arch/amdgpu/cdna5/addr_calc.cpp (hunk 1 above)
Comment (kuhar) [reply emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/isa/arch/amdgpu/cdna5/addr_calc.cpp]: Addressed in becb9554ea9b. `async_lds_lane_address` now takes the access width and rejects the request when `relative_addr + access_size_bytes > wf.lds_size()`. The generated 1/4/8/16-byte async load/store paths pass their actual widths, with a regression test covering a 4-byte-valid/16-byte-invalid boundary case.
